store/postgres/src/store.rs

```diff
@@ -327,84 +541,86 @@ impl BasicStore for Store {
                     }).collect()
             }).map_err(|_| ())
     }
-}
 
-impl BlockStore for Store {
-    fn add_network_if_missing(
+    fn set_block_ptr_with_no_changes(
         &self,
-        new_network_name: &str,
-        new_net_version: &str,
-        new_genesis_block_hash: H256,
+        subgraph_id: SubgraphId,
+        block_ptr_from: EthereumBlockPointer,
+        block_ptr_to: EthereumBlockPointer,
     ) -> Result<(), Error> {
-        use db_schema::ethereum_networks::dsl::*;
+        let conn = self.conn.lock().unwrap();
+        self.update_subgraph_block_pointer(
+            &*conn,
+            subgraph_id.as_str(),
+            block_ptr_from,
+            block_ptr_to,
+        )
+    }
 
-        let network_identifiers_opt = ethereum_networks
-            .select((net_version, genesis_block_hash))
-            .filter(name.eq(new_network_name))
-            .first::<(Option<String>, Option<String>)>(&*self.conn.lock().unwrap())
-            .optional()?;
+    fn transact_block_operations(
+        &self,
+        subgraph_id: &str,
+        block_ptr_from: EthereumBlockPointer,
+        block_ptr_to: EthereumBlockPointer,
+        operations: Vec<EntityOperation>,
+    ) -> Result<(), Error> {
+        // Fold the operations of each entity into a single one
+        let operations = EntityOperation::fold(&operations);
 
-        match network_identifiers_opt {
-            // Network is missing in database
-            None => {
-                insert_into(ethereum_networks)
-                    .values((
-                        name.eq(new_network_name),
-                        head_block_hash.eq::<Option<String>>(None),
-                        head_block_number.eq::<Option<i64>>(None),
-                        net_version.eq::<Option<String>>(Some(new_net_version.to_owned())),
-                        genesis_block_hash
-                            .eq::<Option<String>>(Some(format!("{:x}", new_genesis_block_hash))),
-                    )).on_conflict(name)
-                    .do_nothing()
-                    .execute(&*self.conn.lock().unwrap())?;
-            }
+        let conn = self.conn.lock().unwrap();
 
-            // Network is in database and has identifiers
-            Some((Some(last_net_version), Some(last_genesis_block_hash))) => {
-                if last_net_version != new_net_version {
-                    panic!(
-                        "Ethereum node provided net_version {}, \
-                         but we expected {}. Did you change networks \
-                         without changing the network name?",
-                        new_net_version, last_net_version
-                    );
-                }
+        conn.transaction::<(), _, _>(|| {
```

Comment:
We might need to think about transaction isolation levels here in order to avoid race conditions
https://www.postgresql.org/docs/9.6/static/transaction-iso.html